cs/samples/ReadAddress/ReadAddressApp.cs

```diff
@@ -0,0 +1,206 @@
+                if (arg.ToLower() == ReadCacheArg)
+                {
+                    useReadCache = true;
+                    continue;
+                }
+                if (arg.ToLower() == CheckpointsArg)
+                {
+                    useCheckpoints = true;
+                    continue;
+                }
+                if (arg.ToLower() == RMWArg)
+                {
+                    useRMW = true;
+                    continue;
+                }
+                Console.WriteLine($"Unknown option: {arg}");
+                Usage();
+                return -1;
+            }
+
+            var (store, log, path) = CreateStore();
+            PopulateStore(store);
+            
+            const int keyToScan = 42;
+            ScanStore(store, keyToScan);
+            var cts = new CancellationTokenSource();
+            await ScanStoreAsync(store, keyToScan, cts.Token);
+
+            // Clean up
+            store.Dispose();
+            log.Dispose();
+
+            // Delete the created files
+            try { new DirectoryInfo(path).Delete(true); } catch { }
+
+            Console.WriteLine("Press <ENTER> to end");
+            Console.ReadLine();
+            return 0;
+        }
+
+        private static (FasterKV<Key, Value>, IDevice, string) CreateStore()
+        {
+            var path = Path.GetTempPath() + "FasterReadAddressSample\\";
+            var log = Devices.CreateLogDevice(path + "hlog.log");
+
+            var logSettings = new LogSettings
+            {
+                LogDevice = log,
+                ObjectLogDevice = new NullDevice(),
+                ReadCacheSettings = useReadCache ? new ReadCacheSettings() : null,
+                // Use small-footprint values
+                PageSizeBits = 12, // (4K pages)
+                MemorySizeBits = 20 // (1M memory for main log)
+            };
+
+            var store = new FasterKV<Key, Value>(
+                size: 1L << 20,
+                logSettings: logSettings,
+                checkpointSettings: new CheckpointSettings { CheckpointDir = path },
+                serializerSettings: null,
+                comparer: new Key.Comparer()
+                );
+            return (store, log, path);
+        }
+
+        private static void PopulateStore(FasterKV<Key, Value> store)
+        {
+            // Start session with FASTER
+            using var s = store.For(new Functions()).NewSession<Functions>();
+            Console.WriteLine($"Writing {numKeys} keys to FASTER", numKeys);
+
+            Stopwatch sw = new Stopwatch();
+            sw.Start();
+            var prevLap = 0;
+            for (int ii = 0; ii < numKeys; ii++)
+            {
+                // lap is used to illustrate the changing values
+                var lap = ii / keyMod;
+
+                if (useCheckpoints && lap != prevLap)
+                {
+                    store.TakeFullCheckpoint(out _, CheckpointType.FoldOver);
```

Comment:
Use `await store.TakeFullCheckpointAsync` so that this checkpoint is completed before the next one is initiated.